### PR TITLE
WIP: add support for ST 422:2019

### PR DESCRIPTION
**Pull request by palemieux**

### CODE REVIEW

File: src/MXFTypes.h [hunk 1]
```diff
@@ -688,29 +689,19 @@ namespace ASDCP
 	  const char* EncodeString(char* str_buf, ui32_t buf_len) const;
 	};
 
+      //
       class J2KExtendedCapabilities : public Kumu::IArchive
         {
         public:
-	  i8_t Pcap;
-	  i8_t Ccapi[16]; // this is certainly wrong, need the spec
+	  ui8_t Pcap; // TODO: is this the right type?
```
Comment (palemieux) [inline src/MXFTypes.h]: @jhursty ui32_t

Comment (jhursty) [reply src/MXFTypes.h]: @palemieux palemieux Big-endian, yes?

Comment (palemieux) [reply src/MXFTypes.h]: `(Pcap >> 31) & 0x1` is `Pcap(1)`

See Table A.11ter at https://www.itu.int/rec/T-REC-T.800-201906-I/en

Comment (palemieux) [reply src/MXFTypes.h]: > Big-endian, yes?

It should be written to MXF like any 32-bit integer, so, yes, big endian.

**Comment by jhursty** [comment]
Okay, I think what's checked in is enough for now (i'm only testing against zero so the fact that it's the high bit that is set should not affect the current implementation.)

I'm driving back to L.A. today so there will be radio silence until late this evening.


Best,

-jh

**Comment by palemieux** [comment]
Basically working now. Two suggested next steps:

- figure out how to signal optional parameters at `PictureDescriptor` (@jhursty, any thoughts?)
- add unit test for this revised J2K feature

**Comment by jhursty** [comment]
> figure out how to signal optional parameters at PictureDescriptor 
More CLI options for as-02-wrap, preferably using the --option idiom.

**Comment by palemieux** [comment]
> More CLI options for as-02-wrap, preferably using the --option idiom.

@jhursty I meant that the `PictureDescriptor` structure at https://github.com/cinecert/asdcplib/blob/dd3acfff947ca52e1a3260243cb38f7035626282/src/AS_DCP.h#L1112 does not support optional parameter.

**Comment by jhursty** [comment]
Ah ha. My preference would be for simplicity. In this case, an additional boolean member that signals the presence of these extensions. Multiple boolean members if additional granularity is needed.

Or could you simply switch on the value of ExtendedCapabilities_t.Pcap?

**Comment by palemieux** [comment]
@jhursty @dcbullock What is the process for merging the PR? Do you perform independent testing and/or a code review? Just let me know.

**Comment by palemieux** [comment]
Attached is a J2K codestream that contains CPF and CAP market segments.

[htj2k-codestream.zip](https://github.com/cinecert/asdcplib/files/3975265/htj2k-codestream.zip)

**Pull request merged**